### ISSUE TITLE
Bug in XamForms ViewModelViewHost

### ISSUE THREAD

**Issue by kentcb** [reactiveui/ReactiveUI#891]
This line in the XamForms `ViewModelViewHost` (note: it appears twice in two separate places) causes an exception:

```C#
foreach (View v in this.Children) this.Children.Remove(v);
```
The exception:
```
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
  at System.ThrowHelper.ThrowInvalidOperationException (ExceptionResource resource) [0x00000] in /Users/builder/data/lanes/1926/6916d0e3/source/mono/external/referencesource/mscorlib/system/throwhelper.cs:94
  at System.Collections.Generic.List`1+Enumerator[Xamarin.Forms.Element].MoveNextRare () [0x00016] in /Users/builder/data/lanes/1926/6916d0e3/source/mono/external/referencesource/mscorlib/system/collections/generic/list.cs:1232
  at System.Collections.Generic.List`1+Enumerator[Xamarin.Forms.Element].MoveNext () [0x00050] in /Users/builder/data/lanes/1926/6916d0e3/source/mono/external/referencesource/mscorlib/system/collections/generic/list.cs:1226
  at System.Linq.Enumerable+<CreateWhereIterator>c__Iterator1E`1[Xamarin.Forms.Element].MoveNext () [0x0006f] in //Library/Frameworks/Xamarin.iOS.framework/Versions/8.10.1.74/src/mono/mcs/class/System.Core/System.Linq/Enumerable.cs:3115
  at System.Linq.Enumerable+<CreateOfTypeIterator>c__IteratorC`1[Xamarin.Forms.View].MoveNext () [0x00069] in //Library/Frameworks/Xamarin.iOS.framework/Versions/8.10.1.74/src/mono/mcs/class/System.Core/System.Linq/Enumerable.cs:2154
  at ReactiveUI.XamForms.ViewModelViewHost+<ViewModelViewHost>c__AnonStorey0.<>m__4 (<>__AnonType0`2 x) [0x00131] in /Users/kent/Repository/ReactiveUI/ReactiveUI.XamForms/XamForms/ViewModelViewHost.cs:88
<snip>
```
I replaced with this and it worked (though I haven't taken the time to verify whether things could be better in general):
```C#
while (this.Children.Count > 0)
{
    this.Children.RemoveAt(0);
}
```
I also tried:
```C#
this.Children.Clear();
```
And that also worked. However, I suspect children were being removed one at a time for change notification reasons. Again, haven't bothered to verify at this point.

*And another thing*, the `as View` cast towards the end of the method should really be a regular cast. Otherwise one can get a NRE instead of a cast exception. The latter is slightly more decipherable, but even better would be to remove the need for the cast entirely (if possible).

**Comment by Dessix** [comment]
A quick solution that won't modify whatever the intended behavior was would be a call to IEnumerable.ToArray on this.Children inside the loop declaration; C# doesn't seem to offer a great solution for mid-iteration removal, unlike Java.

**Comment by kentcb** [comment]
Fixed by #919
